support-data-rest/src/main/java/org/uniprot/api/keyword/request/KeywordRequestDTO.java

```diff
@@ -12,24 +12,35 @@
 import org.uniprot.api.rest.validation.ValidSolrSortFields;
 import org.uniprot.store.search.field.KeywordField;
 
+import ebi.ac.uk.uniprot.openapi.extension.ModelFieldMeta;
+import io.swagger.v3.oas.annotations.Parameter;
+
 @Data
 public class KeywordRequestDTO implements SearchRequest {
 
+    @Parameter(description = "Criteria to search the keywords. It can take any valid solr query.")
+    @ModelFieldMeta(path = "support-data-rest/src/main/resources/keyword_query_param_meta.json")
     @NotNull(message = "{search.required}")
     @ValidSolrQuerySyntax(message = "{search.invalid.query}")
     @ValidSolrQueryFields(
             fieldValidatorClazz = KeywordField.Search.class,
             messagePrefix = "search.keyword")
     private String query;
 
+    @Parameter(description = "Name of the field to be sorted on")
+    @ModelFieldMeta(path = "support-data-rest/src/main/resources/keyword_sort_param_meta.json")
     @ValidSolrSortFields(sortFieldEnumClazz = KeywordField.Sort.class)
     private String sort;
 
+    @Parameter(hidden = true)
     private String cursor;
 
+    @Parameter(description = "Comma separated list of fields to be returned in response")
+    @ModelFieldMeta(path = "support-data-rest/src/main/resources/keyword_return_field_meta.json")
     @ValidReturnFields(fieldValidatorClazz = KeywordField.ResultFields.class)
     private String fields;
 
+    @Parameter(description = "Size of the result. Defaults to 25")
```

Comment:
I am not sure if we can do it. Because as you know, we read/scan the code during compile time and it reads the string literal. It cannot replace the variable with value. If you want i can file a Jira ticket to investigate this further.